### PR TITLE
Partial migration from Ant to Gradle

### PR DESCRIPTION
**Pull request by drurenia**
- I started with the easy tasks and migrated few tasks to Gradle: all, upload, checkstyle, clean;
- Added a test task as well to make it easier to execute it (not working because it lacks the venv as specified below);
- Upload and test (newly added) tasks are not working for now because a virtual env is needed to execute them. I haven't figured out yet how to achieve this on Gradle so any suggestion would be great;
- Added Gradle wrapper to eliminate the need for the installation; 
- Importing all remaining tasks from Ant automatically using ant.importBuild "build.xml", so everything still works fine using both ant and gradle commands;

PS: Nothing was added regarding the Exec tasks to address multiple OS's because the Exec tasks are still not working propertly as mentioned above.

Addresses #538

### CODE REVIEW

**Comment by phildini** [comment]
Hi there! Sorry for the long delay in feedaback. What would you suggest is the best way to test this? Can these be run locally easily?

**Comment by drurenia** [comment]
Not a problem @phildini. Since we have a Gradle wrapper in place, you could easily test it using:
`./gradlew <task_name>`

Please bear in mind that I opened this pull request because @eliasdorneles asked me to but I don't think it is supposed so be merged yet. I think since it's my first contribution to this project he wanted to check whether I am going to the right direction or not.

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    workingDir = "tools"
+    commandLine = ["python", "upload.py"]
+}
+
+task clean(type: Delete) {
+    description = "Clean up"
+    delete = ["build", "dist"]
+}
+
+task test(type: Exec) {
```
Comment (eliasdorneles) [inline build.gradle]: This target should be removed: besides the fact that this is hardcoding a virtualenv path, the gradle build config should concern only the Java resources, not the python stuff.

File: build.xml [hunk 2]
```diff
@@ -43,17 +31,4 @@
     <ant antfile="build-android.xml" target="dist"/>
   </target>
 
-  <target name="all" depends="java, android" />
-
-  <target name="upload" depends="java, android" description="Upload assets to S3">
-    <exec executable="python">
-      <arg value="tools/upload.py"/>
-      <arg value="b${release}"/>
```
Comment (eliasdorneles) [inline build.xml]: This argument seems to be missing in the Gradle version.

**Review by eliasdorneles — CHANGES_REQUESTED** [review]
Hi @drurenia !
Sorry for the looong time to review -- I've been eyeing this PR for a while, but life happened! :)
So, I think this is a good start, but there is a few more things I'd hope to see before this is ready to be included -- ideally, at least we'd be able to do `./gradlew build` and it would compile all the Java stuff with Gradle.
I've made some comments below on some things that are not quite right.
How big of a task do you think it would be to move from Ant to Gradle completely?
Thank you!!

**Comment by phildini** [comment]
Hi there! It looks like this PR might be dead, so we're closing it for now. Feel free to re-open it if you'd like to continue, or think about directing your efforts to https://github.com/beeware/briefcase or https://github.com/beeware/toga. Both of these have more active development right now. 😄

**Pull request closed without merging**